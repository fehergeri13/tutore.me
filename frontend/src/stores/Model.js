import {observable, action} from "mobx";

import Auth from "./Auth";
import Rest from "./Rest";

export default class Model {
    @observable.ref auth = new Auth(this);
    @observable.ref routingStore = undefined;
    @observable.ref rest = new Rest(this);

    @observable isMessageModalOpen = false;
    @observable messageUserId = undefined;
    @observable messageUsername = undefined;


    constructor(routingStore) {
        this.routingStore = routingStore;
    }

    @action openMessageModal(messageUserId, messageUsername) {
        this.isMessageModalOpen = true;
        this.messageUserId = messageUserId;
        this.messageUsername = messageUsername;
    }

    @action closeMessageModal() {
        this.isMessageModalOpen = false;
    }
}