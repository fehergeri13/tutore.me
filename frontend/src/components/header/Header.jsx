import React from 'react';


import "./header.less"

export default class App extends React.Component {
    render() {
        return <div className="header">
            <h1>tutore.me</h1>

            <div className="nav">
                <ul className="links">
                    <li><a href="#">Hirdetések</a></li>
                    <li><a href="#">Új hirdetés</a></li>
                </ul>
            </div>

            <div className="user">
                <ul className="links">
                    <li><a href="#">Üzenetek</a></li>
                    <li><a href="#">Profilom</a></li>
                    <li><a href="#">Kilépés</a></li>
                </ul>
            </div>
        </div>;
    }
}
