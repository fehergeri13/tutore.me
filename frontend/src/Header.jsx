import React from 'react';


import "./header.less"

export default class App extends React.Component {
    render() {
        return <div className="header">
            <h1>tutore.me</h1>

            <div className="nav">
                <a href="#">Hirdetések</a>
                <a href="#">Új hirdetés</a>
            </div>

            <div className="user">
                <a href="#">Üzenetek</a>
                <a href="#">Profilom</a>
                <a href="#">Kilépés</a>
            </div>
        </div>;
    }
}
