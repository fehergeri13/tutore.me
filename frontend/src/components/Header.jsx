import React from "react";

import "./header.less";

export default class Header extends React.Component {
    render() {
        return <div className="header">
            <h1>TUTORE.ME - Korrepetálás hirdetés</h1>
            <p>
                Ingyenes korrepetálás hirdetés!
                <a href="#hirdetesek">Hirdetések</a>
                <a href="#feladas">Hirdetés feladás</a>
            </p>
        </div>
    }
}