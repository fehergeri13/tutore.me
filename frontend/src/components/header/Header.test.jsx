import React from 'react';
import renderer from 'react-test-renderer';
import Header from "./Header";
import {MemoryRouter as Router} from "react-router-dom";
import {Provider} from 'mobx-react'
import Model from "../../stores/Model";

describe('Header', () => {
    it('renders correctly', () => {

        const model = new Model();

        const tree = renderer.create(
            <Provider model={model}><Router><Header /></Router></Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
