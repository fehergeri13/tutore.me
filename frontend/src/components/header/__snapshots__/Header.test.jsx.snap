// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="header"
>
  <h1>
    tutore.me
  </h1>
  <div
    className="nav"
  >
    <ul
      className="links"
    >
      <li>
        <a
          href="#"
        >
          Hirdetések
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          Új hirdetés
        </a>
      </li>
    </ul>
  </div>
  <div
    className="user"
  >
    <ul
      className="links"
    >
      <li>
        <a
          href="#"
        >
          Üzenetek
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          Profilom
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          Kilépés
        </a>
      </li>
    </ul>
  </div>
</div>
`;
